File: Dagger2Sample/app/src/main/java/com/sdev/sampledevdagger2/MainActivity.java
package com.sdev.sampledevdagger2;

import android.app.Activity;
import android.os.Bundle;

/**
 * This Activity acts as the view in MVP and it implements View from the contract
 */
public class MainActivity extends Activity implements MainContract.View {
    MainPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //presenter is initialized
        presenter = new MainPresenter(this);

        //Calling the method in presenter
        presenter.addNumbers(1, 2);

    }

    @Override
    public void result(int result) {
        System.out.println("Result = " + result);
    }
}

File: MVIArchitecture/app/src/main/java/com/outofbound/mviarchitecture/view/MainActivity.kt
package com.outofbound.mviarchitecture.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.outofbound.mviarchitecture.R
import com.outofbound.mviarchitecture.domain.MainViewState
import io.reactivex.Observable

class MainActivity : AppCompatActivity(), MainView {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var s = MainPresenter()
        s.bind(this)
    }

    override fun render(state: MainViewState) {
        when(state) {
            is MainViewState.LoadingState -> renderLoadingState()
            is MainViewState.DataLoadedState -> renderDataLoadedState(state)
            is MainViewState.ErrorState -> renderErrorState(state)
        }
    }

    private fun renderErrorState(state: MainViewState.ErrorState) {
        Toast.makeText(this,"Error",Toast.LENGTH_SHORT).show()
    }

    private fun renderDataLoadedState(state: MainViewState.DataLoadedState) {
        Toast.makeText(this,state.fetchedData,Toast.LENGTH_SHORT).show()
    }

    private fun renderLoadingState() {
        Toast.makeText(this,"Loading!!!!",Toast.LENGTH_SHORT).show()
    }
}
